src/click/core.py

```diff
@@ -2554,7 +2554,6 @@ def __init__(
         if help:
             help = inspect.cleandoc(help)
 
-        default_is_missing = "default" not in attrs
```

Comment:
Thank goodness. If we can actually get rid of this line, it removes one of the barriers I kept running into when consolidating signatures and improving typing.